src/view/abstract-element.ts

```diff
@@ -0,0 +1,25 @@
+import { createElement } from '../render';
+
+export interface TemplateElement {
+  template: string;
+  element: Element;
+  removeElement: () => void;
+}
+
+export abstract class AbstractElement {
```

Comment:
Лучше тут тоже сделать дженерик.
Это лучше чем потом везде тайп кастить через `as`

```suggestion
export abstract class AbstractElement<E extends HTMLElement = HTMLDivElement> {
```